src/script.js

```diff
@@ -0,0 +1,346 @@
+        }
+
+        this.shiftStatus = false;
+        ShiftLeft.classList.remove('active');
+        ShiftRight.classList.remove('active');
+      }
+    } else if (this.language === 'RU' && AllCodesAndKeys[event.target.id][1] !== 'controlButton') {
+      if (this.shiftStatus === false && this.CapsLockStatus === false) {
+        this.printedText.value += AllCodesAndKeys[event.target.id][2];
+      } else if (this.shiftStatus === true && this.CapsLockStatus === false) {
+        this.printedText.value += AllCodesAndKeys[event.target.id][3];
+
+        this.shiftStatus = false;
+        ShiftLeft.classList.remove('active');
+        ShiftRight.classList.remove('active');
+      } else if (this.shiftStatus === false && this.CapsLockStatus === true) {
+        if (!event.target.id.match(/(Digit)[0-9]/) && !event.target.id.match(/.*(ash)/)) {
+          this.printedText.value += AllCodesAndKeys[event.target.id][3];
+        } else {
+          this.printedText.value += AllCodesAndKeys[event.target.id][2];
+        }
+      } else if (this.shiftStatus === true && this.CapsLockStatus === true) {
+        if (!event.target.id.match(/(Digit)[0-9]/) && !event.target.id.match(/.*(ash)/)) {
+          this.printedText.value += AllCodesAndKeys[event.target.id][2];
+          this.shiftStatus = false;
+          ShiftLeft.classList.remove('active');
+          ShiftRight.classList.remove('active');
+        } else {
+          this.printedText.value += AllCodesAndKeys[event.target.id][3];
+          this.shiftStatus = false;
+          ShiftLeft.classList.remove('active');
+          ShiftRight.classList.remove('active');
+        }
+      }
+    } else if (event.target.id === 'CapsLock') {
+      if (this.CapsLockStatus === false) {
+        this.CapsLockStatus = true;
+      } else {
+        this.CapsLockStatus = false;
+      }
+    } else if (AllCodesAndKeys[event.target.id][0] === 'Shift') {
+      if (this.shiftStatus === true) {
+        this.shiftStatus = false;
+        ShiftLeft.classList.remove('active');
+        ShiftRight.classList.remove('active');
+      } else {
+        this.shiftStatus = true;
+      }
+    } else if (event.target.id === 'Backspace') {
+      this.printedText.value = this.printedText.value.slice(0, -1);
+    } else if (event.target.id === 'Enter') {
+      this.printedText.setRangeText('\n', this.printedText.selectionStart, this.printedText.selectionEnd, 'end');
+    } else if (event.target.id.match(/(Arrow).*/)) {
+      this.printedText.value += event.target.innerHTML;
+    }
+  }
+
+  MouseUp(event) {
+    if (event.target.id !== 'CapsLock' && !event.target.id.match(/.*(Shift)/)) {
+      if (event.target.classList !== undefined) {
+        event.target.classList.remove('active');
+      }
+      if (event.target.parentNode.classList !== undefined) {
+        event.target.parentNode.classList.remove('active');
+      }
+    }
+    this.printedText.focus();
+  }
+
+  MouseOut(event) {
```

Comment:
why do we need this?
you can just add `:hover` style for buttons
you should use `blur` event to focus textarea, when it loses the focus

Reply:
If this is re line 292: focus on textare is required: 
A) for correct performance of button Enter. Without focus after pressing 1) real keys 2) virtual keys 3) Enter - the whole line goes down. 
B) to bring back cursor after input by virtual keys. 

If this is re line 295: mouseout event is required to avoid sticky keys (залипание клавиш).

If desribed issues can be solved by :hover and blur and these ways are preferable, please, additional explanation is needed. Thanks in advance.